dhall/src/Dhall/Parser/Token.hs

```diff
@@ -349,29 +352,32 @@ hexNumber = choice [ hexDigit, hexUpper, hexLower ]
       where
         predicate c = 'a' <= c && c <= 'f'
 
-lineComment :: Parser ()
+lineComment :: Parser Text
 lineComment = do
     _ <- text "--"
 
     let predicate c = ('\x20' <= c && c <= '\x10FFFF') || c == '\t'
 
-    _ <- Dhall.Parser.Combinators.takeWhile predicate
+    commentText <- Dhall.Parser.Combinators.takeWhile predicate
 
     endOfLine
 
-    return ()
+    return ("--" <> commentText)
   where
     endOfLine =
         (   void (Text.Parser.Char.char '\n'  )
         <|> void (Text.Parser.Char.text "\r\n")
+        <|> Text.Megaparsec.eof
```

Comment:
This line is added because of this: https://github.com/dhall-lang/dhall-lang/issues/1046, we can rollback it if necessary